sphinxlint.py

```diff
@@ -255,11 +268,62 @@ def paragraphs(lines):
         yield paragraph_lno, "".join(paragraph)
 
 
+QUOTE_PAIRS = [
+    '»»',  # Swedish
+    '‘‚',  # Albanian/Greek/Turkish
+    '’’',  # Swedish
+    '‚‘',  # German
+    '‚’',  # Polish
+    '“„',  # Albanian/Greek/Turkish
+    '„“',  # German
+    '„”',  # Polish
+    '””',  # Swedish
+    '››',  # Swedish
+    "''",  # ASCII
+    '""',  # ASCII
+    "<>",  # ASCII
+    "()",  # ASCII
+    "[]",  # ASCII
+    "{}",  # ASCII
+]
+
+QUOTE_PAIRS_NEGATIVE_LOOKBEHIND = "(?<!" + (
+    "|".join(f"{re.escape(left)}`{re.escape(right)}" for left, right in QUOTE_PAIRS) +
+    "|" +
+    "|".join(f"{re.escape(openers[i])}`{re.escape(closers[i])}" for i in range(len(openers)))
```

Comment:
This can be implemented using `zip()`:
```suggestion
    "|".join(f"{re.escape(opener])}`{re.escape(closer)}" for opener, closer in zip(openers, closers))
```
Same as above but on two lines:
```suggestion
    "|".join(f"{re.escape(opener])}`{re.escape(closer)}"
             for opener, closer in zip(openers, closers))
```
With `re.escape` before the `zip`ping:
```suggestion
    "|".join(f"{opener}`{closer}" for opener,closer in zip(map(re.escape, openers),
                                                           map(re.escape, closers)))
```
Adding a comment explaining what is this regex for would also be useful.